Source/Atomic/UI/UIWidget.h

```diff
@@ -203,6 +204,17 @@ class ATOMIC_API UIWidget : public Object, public tb::TBWidgetDelegate
 
     void DeleteAllChildren();
 
+    /// searches for specified widget ID from the top of the widget tree, returns the 1st one found.
+    virtual UIWidget *FindWidget ( const String& searchid );
+    /// return all of the widgets of the specified classname and id that is not 0 from the current widget
+    virtual void SearchWidgetClass ( const String& className, UISelectItemSource *results );  
+    ///  return all of the widgets of the specified id and id that is not 0 from the current widget
+    virtual void SearchWidgetId ( const String& searchid, UISelectItemSource *results );
+    /// return all of the widgets with the specified text and id that is not 0 from the current widget
+    virtual void SearchWidgetText ( const String& searchText, UISelectItemSource *results );
+    /// print out the widget tree to stdout from the current widget
```

Comment:
On the Atomic side of things, we should be returning a Vector of UIWidgets here.  It should be possible to fill one of these pretty easily from the tb side of the search.  

I'd rather not have UISelectItemSource being used as a general container, or add a UISelectItemSource dependency to UIWidget when it is ultimately a Vector of Widgets that we are interested in.  

I can look into this later in the week if needed.  Cool?